tests/bdd/zfs.feature

```diff
@@ -139,4 +139,133 @@ Feature: Validate volume provisioning for fsType zfs, ext4, xfs, btrfs
+    And pvc is created referencing this storage class with a deployment using the same pvc
+    And a zfsvolume is created and used by the application
+    When shared set to "yes" in storage class
+    Then the LocalPV-ZFS Driver will allow the volumes to be mounted by more than one pods
+    Examples:
+      | fsType |
+      |  zfs   |
+      |  ext4  |
+      |  xfs   |
+      |  btrfs |
+
+  Scenario Outline: Supported compression algorithm
+    Given a storage class is created with fsType as zfs, ext4, xfs, btrfs
+    And pvc is created referencing this storage class with a deployment using the same pvc
+    When the zfsvolume compression is present 
+    Then the system must support the <compression_type> algorithm
+    Examples:
+      | compression_type|
+      |     lzjb        |  
+      |     zstd        |
+      |     zstd-1      |
+      |     zstd-2      |
+      |     zstd-3      |
+      |     zstd-4      |
+      |     zstd-5      |
+      |     zstd-7      |
+      |     zstd-8      |
+      |     zstd-9      |
+      |     zstd-10     |
+      |     zstd-11     |
+      |     zstd-12     |
+      |     zstd-13     |
+      |     zstd-14     |
+      |     zstd-15     |
+      |     zstd-16     |
+      |     zstd-17     |
+      |     zstd-18     |
+      |     zstd-19     |
+      |     gzip        | 
+      |     gzip-1      |
+      |     gzip-2      |
+      |     gzip-3      |
+      |     gzip-4      |
+      |     gzip-5      |
+      |     gzip-6      |
+      |     gzip-7      |
+      |     gzip-8      |
+      |     gzip-9      |
+      |     zle         |
+      |     lz4         | 
+
+############################################################################################################
+################## The below bdd need to be implemented in e2e tests ########################################
+############################################################################################################
+
+  Scenario Outline: Allowed topologies 
+    Given a storage class with allowedTopologies is created with fsType as <fsType>
+    And pvc is created referencing this storage class with a deployment using the same pvc
+    And a zfsvolume is created and used by the application
+    When the topology for specific zpool is defined in the storage class
+    Then the volumes must be provisioned on the nodes which has the required zpool
+    Examples:
+      | fsType |
+      |  zfs   |
+      |  ext4  |
+      |  xfs   |
+      |  btrfs |
+
+   Scenario Outline: Import Existing Volumes to LocalPV-ZFS
+    Given a storage class is created with fsType as <fsType>
+    And pvc is created referencing this storage class with a deployment using the same pvc
+    And a zfsvolume is created and used by the application
+    When a node fails, with all the volumes volumes and their data intact as disks are still in good shape
+    Then the disks can be moved to the other node and import the pool there
```

Comment:
When the disks is attached to a different node
Then the zfs pool should be imported on that pool
And the zfsvolume resource should reflect that on creation of a volume referencing the pool on that node
And the application should keep running

Now checking the data loss part might be difficult, we might need some real data base etc to validate that probably.